File: onyx-database/src/main/java/com/onyx/fetch/TableScanner.java
package com.onyx.fetch;

import com.onyx.exception.OnyxException;

import java.util.Map;

/**
 * Created by timothy.osborn on 1/6/15.
 *
 * Contact used to scan entities
 */
public interface TableScanner
{

    /**
     * Full scan
     *
     * @return Matching references meeting criteria
     * @throws OnyxException Cannot scan entity
     */
    Map<Long, Long> scan() throws OnyxException;

    /**
     * Scan with indexes
     *
     * @param existingValues Existing references to start from
     * @return Matching references meeting criteria
     * @throws OnyxException Cannot scan entity
     */
    Map<Long, Long> scan(Map<Long, Long> existingValues) throws OnyxException;

}

File: onyx-database/src/main/kotlin/com/onyx/interactors/scanner/TableScanner.kt
package com.onyx.interactors.scanner

import com.onyx.exception.OnyxException
import com.onyx.interactors.query.QueryCollector
import com.onyx.interactors.record.data.Reference

/**
 * Created by timothy.osborn on 1/6/15.
 *
 * Contact used to scan entities
 */
interface TableScanner {

    var isLast:Boolean
    var collector: QueryCollector<Any>?

    /**
     * Full scan
     *
     * @return Matching references meeting criteria
     * @throws OnyxException Cannot scan entity
     */
    @Throws(OnyxException::class)
    fun scan(): MutableSet<Reference>

    /**
     * Scan with indexes
     *
     * @param existingValues Existing references to start from
     * @return Matching references meeting criteria
     * @throws OnyxException Cannot scan entity
     */
    @Throws(OnyxException::class)
    fun  scan(existingValues: Set<Reference>): MutableSet<Reference>

}
